pytorch_lightning/plugins/plugin_connector.py

```diff
@@ -94,3 +100,80 @@ def __attach_cluster(self, limit=1):
+        """
+        if isinstance(plugins, str):
+            return [self._convert_str_to_plugin(plugins)]
+        return [self._convert_str_to_plugin(plugin) for plugin in plugins]
+
+    def _convert_str_to_plugin(self, plugin):
+        if isinstance(plugin, str):
+            if plugin not in LightningCustomPlugins.__members__:
+                raise MisconfigurationException(
+                    f"{plugin} is not a supported lightning custom plugin. "
+                    f"If you're trying to pass a custom plugin, please pass this as an object to "
+                    f"Trainer(plugins=[MyPlugin()]."
+                    f"Supported plugins as string input: "
+                    f"{(e.name for e in LightningCustomPlugins)}."
+                )
+            plugin_cls = LightningCustomPlugins[plugin].value
+            return plugin_cls()
+        return plugin
+
+    def _add_required_plugin_combinations(self,
+                                          plugins: List[Union[DDPPlugin, PrecisionPlugin, ClusterEnvironment]]):
+        """
+            Allows custom plugins to define additional plugins. This is useful for when custom plugins
+            need to enforce override of native amp/apex when they are enabled.
+        Args:
+            plugins: List of plugins
+
+        Returns: List of plugins containing additional plugins if needed.
+
+        Example::
+            class MyPlugin(DDPPlugin):
+                def required_plugins(self):
+                    return [MyCustomAMPPlugin()]
+
+            # Will automatically add the necessary AMP plugin
+            trainer = Trainer(plugins=[MyPlugin()])
+
+            # Crash as MyPlugin enforces custom AMP plugin
+            trainer = Trainer(plugins=[MyPlugin(), NativeAMPPlugin()])
+
+        """
+        additional_plugins = []
+        for plugin in plugins:
+            required_plugins = plugin.required_plugins(amp_backend=self.trainer.amp_backend)
+            if required_plugins:
+                rank_zero_warn(
+                    f'plugin {type(plugin)} has added additional required plugins as default: '
+                    f'{[type(x) for x in required_plugins]}'
+                    f'Extend this plugin and override required_plugins if this conflicts with your additional plugins.')
+                additional_plugins += required_plugins
+        return plugins + additional_plugins
+
+    @classmethod
+    def available_plugins(cls):
+        """
+            List of all available plugins that can be string arguments to the trainer.
+            Returns: List of all available plugins that are supported as string arguments.
+        """
+        return [e.name for e in LightningCustomPlugins]
+
+
+class LightningCustomPlugins(Enum):
+    """
+        String support for custom lightning plugins.
+        Allows easier access to custom lightning plugins from the command line.
+    """
+    ddp_sharded = DDPShardedPlugin
```

Comment:
how this allows string when you set the class?
do you mean look-up-table
```
LIGHTNING_CUSTOM_PLUGINS = dict(
    ddp_sharded=DDPShardedPlugin,
    native_amp=NativeAMPPlugin,
    apex_amp=ApexPlugin,
)
```

Reply:
I'd prefer an enum, as this allows the class to define the options neatly. Enum's can be used for string conversions as well, and when things get trickier with combinations I think it would be neater!